ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp

```diff
@@ -201,6 +219,8 @@ int enigma_main(int argc, char** argv) {
     for (auto update_hook : extension_update_hooks)
       update_hook();
 
+    fireEventsFromQueue();
```

Comment:
@RobertBColton Here I added the fire function, kindly check. ENIGMA is building successfully. Btw I will write also the semaphore code implementation for that so if you have any suggestions for the perfect place I would really appreciate it.

Reply:
This worked and we talked about it, nice work handling the message queue!